sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/OperationType.java

```diff
@@ -10,6 +10,7 @@
     AbortPartitionMigration,
     AbortSplit,
     AddComputeGatewayRequestCharges,
+    Batch,
```

Comment:
RxDocumentServiceRequest today has a method isReadOnlyScript and isReadOnlyRequest - they are basically used to decide whether CosnistencyReader or ConsistencyWriter is used - currently you would treat all transactional bacthes as writes - even if they only contain ReadItem calls. I think that is a bug/problem - because it means even reads would always be targeting the master replica. It is possible that .Net has the same issue today - but I think it is worth fixing in Java form the beginning?

Reply:
@kirankumarkolli  / @moderakh  / @kushagraThapar  - what are your thoughts on this?

Reply:
This is something we do know and it's not a bug. It's also in my TODOs, but don't know if customer requirement is for read only batch/bulk. However a nice to have feature. Enabling read should work without any backend changes, but of-course verification is needed.

Reply:
agree that this is not functionally blocking. It becomes relevant only when comparing perf between the custom ReadMany API vs. Bulk with just read-operations. I think it is ok to move forward with this PR without addressing it - needs to be addressed across Java and .Net as a perf improvement at some point